utf8proc.c

```diff
@@ -273,19 +282,38 @@ static utf8proc_bool grapheme_break(int lbc, int tbc) {
     true; // GB999
 }
 
-/* return whether there is a grapheme break between codepoints c1 and c2.
+static utf8proc_bool grapheme_break_extended(int lbc, int tbc, utf8proc_int32_t *state)
+{
+  int lbc_override = lbc;
+  if (state && *state != UTF8PROC_BOUNDCLASS_START)
+    lbc_override = *state;
+  utf8proc_bool break_permitted = grapheme_break_simple(lbc, tbc);
+  if (state) {
```

Comment:
Is it really necessary to have general state like this?  Wouldn't it be enough just to know the previous codepoint before `lbc`?